### DISCUSSION TITLE
Observability

### DISCUSSION THREAD

**Discussion by RealOrangeOne** [RealOrangeOne/django-tasks#12] [Ideas]
It's important to be able to interrogate your task queues, to understand how long tasks are taking, queue depth, and other helpful metrics for SREs.

Currently, this must be done in a backend-specific manner. However, perhaps it would be helpful to have some generic APIs provided as part of `django-tasks` to help parse the backend-specific storage into something more easily consumable?

**Comment by chrisgrande** [comment]
A generic API could be very useful, especially with building out a common Django Admin and exposing these parts if someone wanted to integrate them into their own front end.

**Reply by RealOrangeOne** [reply]
Working out which metrics to expose will be the challenge. Queue depth is an easy one, but perhaps p99s etc too.

Building the metrics is the easy bit. The hard bit is working out what to expose by default. _That's_ why it's not in scope for the initial version.

**Comment by jribbens** [comment]
I agree with this - the lack of visibility of what celery is doing is one of its most annoying misfeatures.

**Comment by hooverdc** [comment]
We could also provide something for OpenTelemetry [0] to hook into for tracing. Or at least monkey-patch.

[0] https://opentelemetry.io/docs/zero-code/python/example/
